cmd/kubeadm/app/cmd/token.go

```diff
@@ -86,10 +88,16 @@ func NewCmdToken(out io.Writer, errW io.Writer) *cobra.Command {
 	tokenCmd.PersistentFlags().BoolVar(&dryRun,
 		"dry-run", dryRun, "Whether to enable dry-run mode or not")
 
-	var usages []string
-	var extraGroups []string
-	var tokenDuration time.Duration
-	var description string
+	cfg := &kubeadmapiext.MasterConfiguration{
+		// KubernetesVersion is not used by bootstrap-token, but we set this explicitly to avoid
+		// the lookup of the version from the internet when executing ConfigFileAndDefaultsToInternalConfig
+		KubernetesVersion: "v1.9.0",
```

Comment:
We shouldn't hardcode versions in prod code. How can we workaround this? Why the lookup is a problem here?

Reply:
The lookup access internet, and this is a problem in some environment.

I agree that we should seek for a better solution, but being this already implemented in few other points in kubeadm, IMO we should accept this also here and open a new issue where discuss how to address this consistently across the codebase

Reply:
IIUC, the call stack is `ConfigFileAndDefaultsToInternalConfig` --> `SetInitDynamicDefaults` --> `NormalizeKubernetesVersion` --> `KubernetesReleaseVersion`. If the kubernetes version is configured, `KubernetesReleaseVersion` will `normalizedBuildVersion` the specified version and return, otherwise it will `fetchFromURL` from the internet.